### PR TITLE
Updated Container-Runtime.md

### PR DESCRIPTION
**Pull request by PranshuSrivastava**
This updates the container-runtimes [page](https://kubernetes.io/docs/setup/production-environment/container-runtimes/ of the website to include information about the Dockershim deprecation and so that the users can take an informed decision. This is in accordance with #28449

### CODE REVIEW

**Review by Debanitrkl — COMMENTED** [review]
/cc @adisky 
LGTM

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 1]
```diff
@@ -22,6 +22,15 @@ Kubernetes, on Linux:
...
 
+Dockershim has been [deprecated](https://kubernetes.io/blog/2020/12/02/dockershim-faq/)
+as of [version 1.20.0](https://github.com/kubernetes/kubernetes/pull/94624)
+and is likely to be dropped of
+by 1.24.0, therefore to use Docker you need to be running an older version of
+Kubernetes. You can check out this [documentation](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-deprecation-affects-you/)
+to understand how this deprecation might affect you. For migrating from dockershim you can follow
+[this](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/_print/).
+A robust guide on migrating from Dockershim to other CRIs will be released soon.
+
```
Comment (adisky) [inline content/en/docs/setup/production-environment/container-runtimes.md:33]:
```suggestion
Dockershim has been [deprecated](https://kubernetes.io/blog/2020/12/02/dockershim-faq/)
as of [version 1.20.0](https://github.com/kubernetes/kubernetes/pull/94624)
and is likely to be dropped of by 1.24.0, therefore to use Docker you need to be running an older version of
Kubernetes. 
To know how this deprecation affects you and how to migrate from dockershim you can follow [this document](https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/_print/).
```

Comment (adisky) [reply content/en/docs/setup/production-environment/container-runtimes.md:33]: Also its better if we add this in `notes` section

Comment (PranshuSrivastava) [reply content/en/docs/setup/production-environment/container-runtimes.md:33]: @adisky, should I create another note section for this or add it in the existing one?

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 2]
```diff
@@ -131,9 +148,9 @@ Install containerd:
 {{< tabs name="tab-cri-containerd-installation" >}}
 {{% tab name="Linux" %}}
 
-1. Install the `containerd.io` package from the official Docker repositories. 
-Instructions for setting up the Docker repository for your respective Linux distribution and 
-installing the `containerd.io` package can be found at 
+1. Install the `containerd.io` package from the official Docker repositories.
```
Comment (adisky) [inline content/en/docs/setup/production-environment/container-runtimes.md:151]: why we are changing these lines?

**Comment by adisky** [comment]
preview is here https://deploy-preview-30169--kubernetes-io-main-staging.netlify.app/docs/setup/production-environment/container-runtimes/
@PranshuSrivastava all the links are not rendering correctly
/cc @SergeyKanzhelev

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 3]
```diff
@@ -22,6 +22,21 @@ Kubernetes, on Linux:
 - [CRI-O](#cri-o)
 - [Docker](#docker)
 
+{{< note >}}
+Dockershim has been [deprecated](
+https://kubernetes.io/blog/2020/12/02/dockershim-faq/)
+as of [version 1.20.0](https://github.com/kubernetes/kubernetes/pull/94624)
+and is likely to be dropped ofby 1.24.0, therefore to use Docker you need to be
```
Comment (Mr-DG-Wick) [inline content/en/docs/setup/production-environment/container-runtimes.md:29]:
```suggestion
and is likely to be dropped off by 1.24.0, therefore to use Docker you need to be
```

**Review by sftim — DISMISSED** [review]
It's important that we communicate this deprecation clearly. I've added some feedback that details my suggestions and concerns.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 4]
```diff
@@ -22,6 +22,21 @@ Kubernetes, on Linux:
...
+as of [version 1.20](
+https://kubernetes.io/blog/2020/12/08/kubernetes-1-20-release-announcement/#dockershim-deprecation)
+and is planned to be removed in by 1.24, therefore to use Docker you need to
+be running an older version of Kubernetes.
+You can check out this [documentation](
+https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/check-if-dockershim-deprecation-affects-you/)
+to understand how this deprecation might affect you. For migrating from
+dockershim you can follow [this](
+https://kubernetes.io/docs/tasks/administer-cluster/migrating-from-dockershim/) 
+document to migrate from dockershim.
```
[the remark this answers is not in the document]
Comment (PranshuSrivastava) [reply content/en/docs/setup/production-environment/container-runtimes.md:37]: @sftim I understood what you are saying.  And about the summary text, should I add it in a seperate PR for more clarity or adding it in this PR itself is fine?

**Comment by PranshuSrivastava** [comment]
@vincepri PTAL

**Comment by SergeyKanzhelev** [comment]
> @vincepri PTAL

You still hasn't fixed the line breaks inside links.

**Comment by reylejano** [comment]
@PranshuSrivastava please update links to be path based as in [@sftim's comment](https://github.com/kubernetes/website/pull/30169/files#r735168782)

**Comment by PranshuSrivastava** [comment]
@reylejano PTAL

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 5]
```diff
@@ -22,6 +22,21 @@ Kubernetes, on Linux:
 - [CRI-O](#cri-o)
 - [Docker](#docker)
 
+{{< note >}}
+Docker container runtime (Dockershim) has been [deprecated](
```
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/setup/production-environment/container-runtimes.md:26]: > for links, we'd prefer keeping them on a single line.

This is still true. Put the line break after the `)` that ends the hyperlink, not after the `(`.

File: content/en/docs/setup/production-environment/container-runtimes.md [hunk 6]
```diff
@@ -22,6 +22,18 @@ Kubernetes, on Linux:
...
+/blog/2020/12/02/dockershim-faq/)
+as of [version 1.20](
+/blog/2020/12/08/kubernetes-1-20-release-announcement/#dockershim-deprecation)
+and is planned to be removed in by 1.24, therefore to use Docker you need to
+be running an older version of Kubernetes.
+Read the [Migrating from dockershim](
+/docs/tasks/administer-cluster/migrating-from-dockershim/) section to learn
+more about migrating to an alternative container runtime.
```
Comment (sftim) [inline content/en/docs/setup/production-environment/container-runtimes.md:34]:
```suggestion
Kubernetes has a special integration with Docker, `dockershim`. This special integration
was formally [deprecated](/blog/2020/12/08/kubernetes-1-20-release-announcement/#dockershim-deprecation)
in the v1.20 release of Kubernetes, and is scheduled to be removed in the v1.24 release.

For more information, read
[Don't Panic: Kubernetes and Docker](/blog/2020/12/02/dont-panic-kubernetes-and-docker/), or
visit [Migrating from dockershim](/docs/tasks/administer-cluster/migrating-from-dockershim/).
```

**Review by sftim — CHANGES_REQUESTED** [review]
@PranshuSrivastava this PR looks like it's kind of mixed up

I suggest that the easy approach, if you're willing to, is to open a new PR that just includes the changes you want to make. These should be based on all the feedback so far.

You don't have to use the exact wording I'm proposing here; that's more of a guide. I hope the suggestion is useful though.

**Comment by PranshuSrivastava** [comment]
@sftim The idea of a new PR looks good. I'll do that.

**Pull request closed without merging**